fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/Quantifier.java

```diff
@@ -0,0 +1,233 @@
+        }
+    }
+
+    /**
+     * Factory method to create an existential quantifier over a given expression reference containing relational
+     * expressions.
+     * @param rangesOver expression reference to {@link RelationalExpression}s
+     * @return a for-each quantifier ranging over the given expression reference
+     */
+    @Nonnull
+    public static Existential existential(@Nonnull final ExpressionRef<? extends RelationalExpression> rangesOver) {
+        return new Existential(rangesOver);
+    }
+
+    /**
+     * Physical quantifier. This kind of quantifier is the conduit between two {@link RecordQueryPlan}s. It does
+     * not have any associated semantics; all semantics and execution details must be subsumed
+     * by the query plans themselves.
+     */
+    @SuppressWarnings("squid:S2160") // sonarqube thinks .equals() and hashCode() should be overwritten which is not necessary
+    public static final class Physical extends Quantifier {
+        private final ExpressionRef<? extends RecordQueryPlan> rangesOver;
+
+        private Physical(@Nonnull final ExpressionRef<? extends RecordQueryPlan> rangesOver) {
+            this.rangesOver = rangesOver;
+        }
+
+        @Override
+        @Nonnull
+        public ExpressionRef<? extends RecordQueryPlan> getRangesOver() {
+            return rangesOver;
+        }
+
+        @Override
+        @Nonnull
+        public String getShorthand() {
+            return "𝓅";
+        }
+    }
+
+    /**
+     * Factory method to create a physical quantifier over a given expression reference containing query plans.
+     * @param rangesOver expression reference to {@link RecordQueryPlan}s
+     * @return a physical quantifier ranging over the given reference
+     */
+    @Nonnull
+    public static Physical physical(@Nonnull final ExpressionRef<? extends RecordQueryPlan> rangesOver) {
+        return new Physical(rangesOver);
+    }
+
+    /**
+     * Factory method to create a physical quantifier over newly created expression reference containing the given
+     * record plan.
+     * @param rangesOverPlan {@link RecordQueryPlan} the new quantifier should range over.
+     * @return a physical quantifier ranging over a new expression reference containing the given expression reference
+     */
+    @Nonnull
+    public static Physical physical(@Nonnull final RecordQueryPlan rangesOverPlan) {
+        return new Physical(GroupExpressionRef.of(rangesOverPlan));
+    }
+
+    /**
+     * Return the reference that the quantifier ranges over.
+     * @return {@link ExpressionRef} this quantifier ranges over
+     */
+    @Nonnull
+    public abstract ExpressionRef<? extends RelationalExpression> getRangesOver();
+
+    /**
+     * Return a short hand string for the quantifier. As a quantifier's semantics is usually quite subtle and should
+     * not distract from expressions. For example, when a data flow is visualized the returned string should be <em>short</em>.
+``
```

Comment:
I'm guessing this was an accidental change due to suggested edits
```suggestion
```

Reply:
done.